uvdat/core/rest/guardian.py

```diff
@@ -0,0 +1,51 @@
+from guardian.shortcuts import get_objects_for_user
+from rest_framework.filters import BaseFilterBackend
+from rest_framework.permissions import SAFE_METHODS, IsAuthenticated
+
+from uvdat.core import models
+
+
+class GuardianPermission(IsAuthenticated):
+    def get_object_queryset(self, obj):
+        if isinstance(obj, models.Project):
+            return obj
+        elif isinstance(models.Dataset):
+            return obj.project_set
+        elif (
+            isinstance(obj.models.Chart)
+            or isinstance(obj, models.SimulationResult)
+            or isinstance(obj, models.DerivedRegion)
+        ):
+            return obj.project
+        elif (
+            isinstance(obj, models.FileItem)
+            or isinstance(obj, models.VectorMapLayer)
+            or isinstance(obj, models.RasterMapLayer)
+            or isinstance(obj, models.Network)
+            or isinstance(obj, models.SourceRegion)
+        ):
+            return obj.dataset.project_set
```

Comment:
I just learned this neat fact about `isinstance`, from the [docs](https://docs.python.org/2/library/functions.html#isinstance):

> If classinfo is a tuple of class or type objects (or recursively, other such tuples), return true if object is an instance of any of the classes or types

```suggestion
        if isinstance(
            obj,
            (
                models.FileItem,
                models.VectorMapLayer,
                models.RasterMapLayer,
                models.Network,
                models.SourceRegion,
            ),
        ):
            return obj.dataset.project_set
```

Reply:
Updated in [f831cf3](https://github.com/OpenGeoscience/uvdat/pull/60/commits/f831cf38eefd71fce0d36d418bb3e6810b6ef144)